process.go

```diff
@@ -390,6 +420,59 @@ func (p *process) eventLoop(ctx context.Context) error {
 	}
 }
 
+// writeHtlcs collects and batch writes htlc resolutions to disk. This operation is
+// performed separately to the main event loop so that slow writes do not hold up other,
+// more critical interception activities.
+func (p *process) writeHtlcs(ctx context.Context) error {
+	var (
+		ticker       = ticker.New(htlcWriteInterval)
+		htlcsToWrite []*resolvedHtlc
+	)
+
+	ticker.Resume()
+	defer ticker.Stop()
+
+	for {
+		select {
+		case htlc := <-p.writeHtlcChan:
+			htlcsToWrite = append(htlcsToWrite, htlc)
+
+		case <-ticker.Ticks():
+			// TODO: path!
+			err := writeHtlcs("htlc_forwarding.csv", htlcsToWrite)
+			if err != nil {
+				return err
+			}
+
+			htlcsToWrite = htlcsToWrite[:0]
+
+		case <-ctx.Done():
+			return ctx.Err()
+		}
+	}
+}
+
+// queueHtlcWrite queues a htlc resolution to be written to disk.
+func (p *process) queueHtlcWrite(ctx context.Context, htlc *resolvedHtlc) {
+	select {
+	case p.writeHtlcChan <- htlc:
+
+	case <-ctx.Done():
+		p.log.Infof("Htlc: %v write not reported", htlc)
+	}
+}
+
+// htlcCompleted reports a htlc to the main event loop to supplement it with additional
+// data before it is written to disk.
+func (p *process) htlcCompleted(ctx context.Context, htlc *htlcInfo) {
+	select {
+	case p.completedHtlcChan <- htlc:
+
+	case <-ctx.Done():
+		p.log.Infof("Htlc: %v not reported", htlc)
```

Comment:
Is this a useful log line? If the process exists, there is probably a lot more that isn't reported anymore?